python/taichi/lang/matrix.py

```diff
@@ -19,16 +19,36 @@ class Matrix:
     is_taichi_class = True
 
     def __init__(self,
-                 n,
+                 n=1,
                  m=1,
                  dt=None,
                  empty=False,
                  shape=None,
                  layout=None,
                  needs_grad=False,
-                 keep_raw=False):
+                 keep_raw=False,
+                 rows=None,
+                 cols=None):
         self.grad = None
-        if isinstance(n, list):
+        if rows is not None or cols is not None:
```

Comment:
Let's be slightly more defensive here: Raise a `ValueError` if
 - `rows` and `cols` present simultaneously;
 - elements of `rows` or `cols` do not have the same shape;
 - elements of `rows` or `cols` do not have the `m == 1`.